code/go/0chain.net/smartcontract/storagesc/readpool.go

```diff
@@ -61,19 +58,36 @@ func (rp *readPool) Decode(p []byte) error {
 	return json.Unmarshal(p, rp)
 }
 
-// save the pool in tree
-func (rp *readPool) save(sscKey, clientID string, balances cstate.StateContextI) (
-	err error) {
+func (rp *readPool) add(isOwner bool, balance state.Balance) {
+	if isOwner {
+		rp.OwnerBalance += balance
+		return
+	}
+	rp.VisitorBalance += balance
+}
+
+func (rp *readPool) drain(isOwner bool) (diff state.Balance) {
+	if isOwner {
+		diff = rp.OwnerBalance
+		rp.OwnerBalance = 0
+	} else {
```

Comment:
can avoid using else, just like in add()

Reply:
done